Badge documentation (#1956)
telerik/blazor-docs: components/badge/overview.md

---
title: Overview
page_title: Badge Overview
description: Discover the Badge component for Blazor. Learn how to add the component to your app and explore its configuration options, such as positioning and alignment.
slug: badge-overview
tags: telerik,blazor,badge,navbar
published: True
position: 0
---

# Blazor Badge Overview

> The <a href = "https://www.telerik.com/blazor-ui/badge" target="_blank">Blazor Badge component</a> is a visual element that displays additional information, such as status indicators, notification icons, short text, and others.

Reviewer: Probably singular is more precise, because a single Badge displays one indicator / icon, right?

```suggestion
The <a href = "https://www.telerik.com/blazor-ui/badge" target="_blank">Blazor Badge component</a> is a visual element that displays additional information, such as a status indicator, notification icon, short text, and others.
```

> The Badge comes with built-in customization features that allow the developer to change the [background color]({%slug badge-appearance%}#themecolor) and the [fill mode]({%slug badge-appearance%}#fillmode), to enable or disable the cutout border, and fine-tune the [positioning and alignment]({%slug badge-position-alignment%}) of the component.

Reviewer: Sort the features by importance. Positioning and alignment have a higher value than appearance settings.

## Creating Blazor Badge

1. Add the `<TelerikBadge>` tag to a container that the Badge will attach to.

> 1. Use the `ChildContent` to add content.

Author: Its not a property, it is the space between the opening and closing tag of the component.

> 1. Use the `ChildContent` to add content.

Reviewer 2: > Its not a property, it is the space between the opening and closing tag of the component.

If there is no appropriate noun that you can add after `ChildContent`, I suggest removing the definite article:
``1. Use `ChildContent` to add content.``

> 1. Use the `ChildContent` to add content.

Reviewer: The instruction is ambiguous.
```suggestion
1. Use the Badge `ChildContent` to add content.
```

1. (optional) Customize the [appearance]({%slug badge-appearance%}) and fine-tune the [positioning and alignment]({%slug badge-position-alignment%}) of the Telerik Blazor Badge.

>caption Basic configuration of the Blazor Badge

````CSHTML
<TelerikButton>
    <TelerikBadge>12</TelerikBadge>

>     <span>Notifications</span>

Reviewer: The `<span>` doesn't seem to be required?
```suggestion
   Notifications
```

</TelerikButton>
````

## Positioning and Alignment

> You can control how the Badge is positioned relative to its associated container. [Read more about the Blazor Badge positioning...]({%slug badge-position-alignment%})

Reviewer: There is no need to optimize our documentation pages for words like "read more about", so they should be outside the links. Overall, it's best if you can integrate links in regular text, rather than "read more" phrases.

```suggestion
You can control how the Badge is positioned relative to its associated container. Read more about the [Blazor Badge positioning...]({%slug badge-position-alignment%})
```

### Styling and Appearance

The following parameters enable you to customize the appearance of the Blazor Badge:

| Parameter | Type | Description |
| --- | --- | --- |
| `Class` | `string` | Defines the `<div class="k-badge">` CSS class rendered on the main wrapping element of the Badge component. Use it for [styling customizations]({%slug themes-override%}). |
| `FillMode` | `Telerik.Blazor.ThemeConstants.Badge.FillMode.Solid` | Controls how the Badge is filled. |
| `Rounded` | `Telerik.Blazor.ThemeConstants.Badge.Rounded.Full` | Defines how rounded the borders of the Badge are. |
| `Size` | `Telerik.Blazor.ThemeConstants.Badge.Size.Medium` | Sets the size of the Badge. |
| `ThemeColor` | `Telerik.Blazor.ThemeConstants.Badge.ThemeColor.Primary` | Adjusts the background color of the Badge. |

You can find more information for customizing the Badge appearance in the [Appearance article]({%slug badge-appearance%}).

## Next Steps

* [Explore the Badge Positioning and Alignment Settings]({%slug badge-position-alignment%})
* [Customize the Badge Appearance]({%slug badge-appearance%})

> * [Add a Badge to a Container that Cannot Have Children (Knowledge-Base Article)]({slug kb-badge-container-no-children})

Reviewer: The slug syntax is invalid. Is this on purpose due to the missing KB? I would add the KB directly to this PR.

Author: I have not yet added the KB, it's in my to-do list for the docs.
